### ISSUE TITLE
Intellij source folder should be marked as generated

### ISSUE THREAD

**Issue by dcabasson** [google/protobuf-gradle-plugin#222]
Currently source folders in the Intellij configuration are not set as generated :
`<sourceFolder url="file://$MODULE_DIR$/src/generated-sources/main/java" isTestSource="false"/>`

See https://www.jetbrains.com/help/idea/creating-and-managing-modules.html#folder-categories for context

**Comment by pgelinas** [comment]
Yeah, that is still marked as "incubating" in the gradle doc. Simple workaround is add it yourself in your gradle build:

```
def generatedJava = file("${protobuf.generatedFilesBaseDir}/main/java")
def generatedGrpc = file("${protobuf.generatedFilesBaseDir}/main/grpc")
idea {
    module {
        generatedSourceDirs += generatedJava
        generatedSourceDirs += generatedGrpc
    }
}
```
![image](https://user-images.githubusercontent.com/261076/40131789-f0b20110-5908-11e8-8bca-e701e4dda845.png)

**Comment by zpencer** [comment]
Right, the only reason they are not marked as generated is because that API is unstable. Out of curiosity, is this causing any issues for your IDE workflow?

**Comment by saturnism** [comment]
quick ping on this issue.  having to mark individual source folders is difficult for a large project w/ multiple proto sources.

**Comment by timothybasanov** [comment]
Would it be possible to add an option to protobuf plugin to optionally use this incubating API?
Incorrectly marked source folders make it harder to navigate bigger projects.

**Comment by timothybasanov** [comment]
@zpencer: As of Gradle 5.0 this is no longer an incubating API, would you consider checking Gradle version and switching to generated sources?
https://docs.gradle.org/5.0/dsl/org.gradle.plugins.ide.idea.model.IdeaModule.html#org.gradle.plugins.ide.idea.model.IdeaModule:generatedSourceDirs

As for UI reasons, it does not break anything per se, but generated sources are shown/sorted differently in UI. It makes it easier to distinguish between "real" application code and "generated".

**Comment by zpencer** [comment]
I don't have cycles to work on this plugin at the moment but PRs are welcome as usual :)

**Comment by timothybasanov** [comment]
@zpencer Would it be ok to just enable it? Or do you want me to check Gradle version and only enable it if 5.x is detected?

**Comment by zpencer** [comment]
I think the safest way is to make sure the gradle version is at least 5.0. There may be users of the plugin at lower versions.

**Comment by voidzcy** [comment]
Closing, this should have been fixed by #477.
